### PR TITLE
Updating IF_RWR_RRD_MRD_CNS and IF_RWR_RRD_SRD_CNS to encode reg2, rather than reg1, in 3456

### PR DESCRIPTION
**Pull request by tannergooding**
This should resolve the other issues in https://github.com/dotnet/coreclr/issues/16132

FYI. @CarolEidt, @fiigii

### CODE REVIEW

**Comment by tannergooding** [comment]
This was already handled elsewhere, but was missed for these two cases in particular.

**Comment by tannergooding** [comment]
test x86_checked_windows_nt_jitstressregs4 please

**Comment by tannergooding** [comment]
test Windows_NT x86 Checked jitstressregs4 please

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -12094,7 +12094,7 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
                 if (IsDstDstSrcAVXInstruction(ins))
```
Comment (CarolEidt) [inline src/jit/emitxarch.cpp]: Are there no cases where we use this format for an `IsDstSrcSrcAvxInstruction()`? I'm having trouble imagining how that would make sense for these formats. Would an assert make sense?

Comment (tannergooding) [reply src/jit/emitxarch.cpp]: We should actually not have this check (or a `IsDstSrcSrcAvxInstruction`) check at all (for these formats).

Those two checks are meant for the case where we call a 2 operand encoding (such as `IF_RRW_MRD_CNS`) and the third operand is implicitly determined.

This particular format is used when the emitter was explicitly given the register for the 2nd operand, so there is no checks needed.

Comment (CarolEidt) [reply src/jit/emitxarch.cpp]: Right, so perhaps we should simply remove it. I think that might be clearer. And perhaps there could be an assert that we're using AVX encodings?

Comment (tannergooding) [reply src/jit/emitxarch.cpp]: Was just working on doing that 😄

Comment (tannergooding) [reply src/jit/emitxarch.cpp]: @CarolEidt, should be fixed now. Also updated `IF_RWR_RRD_MRD` and `IF_RWR_RRD_SRD`, since they were incorrect as well.

**Comment by tannergooding** [comment]
test Windows_NT x86 Checked jitstressregs4 please

**Pull request merged**